src/main/java/randoop/generation/constanttfidf/ConstantMiningStatistics.java

```diff
@@ -0,0 +1,298 @@
+   * @param seq the sequence to be added
+   * @param numClassesWithConstant the number of classes in the current scope that contain the
+   *     sequence to be added
+   */
+  public void addToNumClassesWith(Object type, Sequence seq, int numClassesWithConstant) {
+    Object scope;
+    switch (GenInputsAbstract.literals_level) {
+      case CLASS:
+        throw new RuntimeException("Should not update classesWithConstant in CLASS level");
+      case PACKAGE:
+        scope = ((ClassOrInterfaceType) type).getPackage();
+        break;
+      case ALL:
+        scope = null;
+        break;
+      default:
+        throw new RuntimeException("Unknown literals level");
+    }
+    scopeStatisticsMap
+        .computeIfAbsent(scope, __ -> new ScopeStatistics())
+        .addClassesWith(seq, numClassesWithConstant);
+  }
+
+  /**
+   * Add and update the numClasses of the current scope.
+   *
+   * @param type the type of the class
+   * @param numClasses the total number of classes in the current scope
+   */
+  public void addToTotalClasses(Object type, int numClasses) {
+    Object scope;
+    switch (GenInputsAbstract.literals_level) {
+      case CLASS:
+        throw new RuntimeException("Should not update totalClasses in CLASS level");
+      case PACKAGE:
+        scope = ((ClassOrInterfaceType) type).getPackage();
+        break;
+      case ALL:
+        scope = null;
+        break;
+      default:
+        throw new RuntimeException("Unknown literals level");
+    }
+    scopeStatisticsMap
+        .computeIfAbsent(scope, __ -> new ScopeStatistics())
+        .addToTotalClasses(numClasses);
+  }
+
+  /**
+   * Get all sequences that had been recorded under the specific scope, which are the constants
+   * extracted by constant mining.
+   *
+   * @param scope the specific package, class, or null
+   * @return the set of sequences that have been recorded under the specific scope
+   */
+  public Set<Sequence> getSequencesForScope(Object scope) {
+    ScopeStatistics stats = scopeStatisticsMap.get(scope);
+    if (stats == null) {
+      Log.logPrintf("The scope %s is not found in the frequency information", scope);
+      return new HashSet<>();
+    }
+
+    return stats.getSequenceSet();
+  }
+
+  /**
+   * Get the frequency information for all scopes.
+   *
+   * @return the frequency information for all scopes
+   */
+  public Map<Object, Map<Sequence, Integer>> getNumUses() {
+    Map<Object, Map<Sequence, Integer>> res = new HashMap<>();
+    scopeStatisticsMap.forEach((key, value) -> res.put(key, value.getNumUses()));
+    return res;
+  }
+
+  /**
+   * Get the frequency information of the given scope.
+   *
+   * @param scope a type, a package, or null
+   * @return the frequency information of the given scope
+   */
+  public Map<Sequence, Integer> getNumUses(Object scope) {
+    switch (GenInputsAbstract.literals_level) {
+      case CLASS:
+        return scopeStatisticsMap.get((ClassOrInterfaceType) scope).getNumUses();
```

Comment:
What is the point of this cast?  I think the whole method body can just be
```java
scopeStatisticsMap.get(scope).getNumUses();
```
without any switch statement, either.